consensus/fork_choice/src/fork_choice.rs

```diff
@@ -1109,6 +1100,25 @@ where
         Ok(())
     }
 
+    pub fn on_attester_slashing(
+        &mut self,
+        slashing: &AttesterSlashing<E>,
+        head_state: &BeaconState<E>,
+    ) -> Result<(), Error<T::Error>> {
+        // We assume that the attester slashing provided to this function has already been verified.
+        //
+        // Instead of loading the justified state as per the spec, we use the head state that the
+        // slashing was verified against. The differences should be negligible, if a validator is
+        // slashable at the justified state but not at the head state then their withdrawable epoch
+        // must fall between justification and the head, which means they are incapable of attesting
+        // anyway. If a validator is slashable at the head but not at justification then it means
+        // their activation has occurred since justification, and we arguably shouldn't be counting
+        // their attestations.
+        let slashed_indices = get_slashable_indices(head_state, slashing)?;
```

Comment:
I think if we wanted to match the spec as closely as possible without actually loading the justified state then we could use `get_slashable_indices_modular` and check `validator.is_slashable_at(head_state.current_justified_checkpoint().epoch)`. But I suspect that the spec is only written like that because the justified state is easy to access without plumbing the head state through. I've asked about it on the spec PR here: https://github.com/ethereum/consensus-specs/pull/2845#discussion_r930496121. We can hold off on merging for a bit until we hear back

Reply:
I was wrong :sweat_smile: 

Fixed in this commit, which also enabled a few other simplifications ad0621e49df3dde19ac7bb3e33efa22265d0671a.